### ISSUE TITLE
[Feature] Upgrade base docker image to dotnet 7

### ISSUE THREAD

**Issue by gaelj** [microsoft/playwright-dotnet#2466]

**Comment by mxschmitt** [comment]
The .NET version inside the Docker images is always the latest LTS version. The latest LTS version is currently .NET 6 see [here](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core).

**Comment by akrock** [comment]
@mxschmitt You already support two different base NET6 tags for `jammy` and `focal`. It doesn't seem like much of a stretch for you to add more builds for NET 7.

Having support for all current NET SDK versions, will increase the flexibility and choices that consumers.

```
# 1. Add tip-of-tree Playwright package to install its browsers.
#    The package should be built beforehand from tip-of-tree Playwright.
COPY ./dist/* /tmp/
```
Maybe trivial with your environments, but step 1 of the process being *build the whole thing from source* makes it unfriendly for people who aren't actively developing on this project to utilize it.

TLDR:
- Level of effort for adding yet another docker tag: `Low`
- Level of effort for consumers to build this project from source and assembly their own docker image for an actively `Supported Language Provided By The Company That Also Owns This Project`: `High++`

**Comment by cgv-wd** [comment]
> @mxschmitt You already support two different base NET6 tags for `jammy` and `focal`. It doesn't seem like much of a stretch for you to add more builds for NET 7.
> 
> Having support for all current NET SDK versions, will increase the flexibility and choices that consumers have.
> 
> ```
> # 1. Add tip-of-tree Playwright package to install its browsers.
> #    The package should be built beforehand from tip-of-tree Playwright.
> COPY ./dist/* /tmp/
> ```
> 
> Maybe trivial with your environments, but step 1 of the process being _build the whole thing from source_ makes it unfriendly for people who aren't actively developing on this project to utilize it.
> 
> TLDR:
> 
> * Level of effort for adding yet another docker tag: `Low`
> * Level of effort for consumers to build this project from source and assemble their own docker image for an actively `Supported Language Provided By The Company That Also Owns This Project`: `High++`

Just wanted to add that I also think this is a good feature request that should be reconsidered, per reasons listed here.  Additionally, the value a low-hanging fruit like this adds should not have it's symbolic value overlooked... Choosing not to include builds for NET 7 discourages consumers from targeting the current _Standard_ library (NET 7) for their projects and only stands to inhibit technical debt. One may argue that 'LTS' is good enough as it's the well established and more mature iteration that is supported w/ a release cycle, but I argue that M$ should be more focused on setting the precedent that their *'Standard'* libraries are supported in their latest frameworks, such as Playwright .NET.

**Comment by gaelj** [comment]
@mxschmitt would you please be so kind as to reconsider your position and support the latest framework in addition to LTS ? See above comments. In my case, this has the consequence of having to maintain a net6 version of our core libraries just because we use playwright in docker, therefore use more storage and increasing CI times. Thanks for your understanding.

**Comment by MrSimonC** [comment]
If you're reading this, as I was, struggling to get .net 7 working with Playwright in Docker (and even though I agree with all the community-mentioned points above) I'm submitting my current workaround `Dockerfile` for those of you who are looking for a working `.net 7.0` `Dockerfile`:

```
# Custom .net 7 image as Playwright team only use .net LTS for docker images: https://github.com/microsoft/playwright-dotnet/issues/2466
# Official build steps for LTS Docker image: https://github.com/microsoft/playwright-dotnet/blob/main/utils/docker/Dockerfile.focal

FROM mcr.microsoft.com/dotnet/sdk:7.0-jammy as base
ENV PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
RUN mkdir /ms-playwright && \
    mkdir /ms-playwright-agent && \
    cd /ms-playwright-agent && \
    dotnet new console && \
    echo '<?xml version="1.0" encoding="utf-8"?><configuration><packageSources><add key="local" value="/tmp/"/></packageSources></configuration>' > nuget.config && \
    dotnet add package Microsoft.Playwright && \
    dotnet build && \
    ./bin/Debug/net7.0/playwright.ps1 install --with-deps && \
    rm -rf /var/lib/apt/lists/* && \
    rm -rf /tmp/* && \
    rm -rf /ms-playwright-agent && \
    chmod -R 777 /ms-playwright

FROM mcr.microsoft.com/dotnet/sdk:7.0-jammy AS build
WORKDIR /src
COPY ["MyProject/MyProject.csproj", "./MyProject/"]
COPY ["Shared/Shared.csproj", "./Shared/"]
RUN dotnet restore "./MyProject/MyProject.csproj"
COPY . .
RUN dotnet build "./MyProject/MyProject.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "./MyProject/MyProject.csproj" -c Release -o /app/publish /p:UseAppHost=false

FROM base AS final
EXPOSE 5000
ENV ASPNETCORE_URLS=http://+:5000
WORKDIR /app
COPY --from=publish /app/publish .
ENTRYPOINT ["dotnet", "MyProject.dll"]

```

**Comment by bokece** [comment]
Just an enhancement to ⭐ @MrSimonC ⭐ response in order to reduce the final image size. 

```
# Custom .net 7 image as Playwright team only use .net LTS for docker images: https://github.com/microsoft/playwright-dotnet/issues/2466
# Official build steps for LTS Docker image: https://github.com/microsoft/playwright-dotnet/blob/main/utils/docker/Dockerfile.focal

FROM mcr.microsoft.com/dotnet/sdk:7.0-jammy as base
ENV PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
RUN mkdir /ms-playwright && \
    mkdir /ms-playwright-agent && \
    cd /ms-playwright-agent && \
    dotnet new console && \
    echo '<?xml version="1.0" encoding="utf-8"?><configuration><packageSources><add key="local" value="/tmp/"/></packageSources></configuration>' > nuget.config && \
    dotnet add package Microsoft.Playwright && \
    dotnet build && \
    ./bin/Debug/net7.0/playwright.ps1 install --with-deps && \
    rm -rf /var/lib/apt/lists/* && \
    rm -rf /tmp/* && \
    rm -rf /ms-playwright-agent && \
    chmod -R 777 /ms-playwright

FROM mcr.microsoft.com/dotnet/sdk:7.0-jammy AS build
WORKDIR /src
COPY ["MyProject/MyProject.csproj", "./MyProject/"]
COPY ["Shared/Shared.csproj", "./Shared/"]
RUN dotnet restore "./MyProject/MyProject.csproj"
COPY . .
RUN dotnet build "./MyProject/MyProject.csproj" -c Release -o /app/build

FROM build AS publish
RUN dotnet publish "./MyProject/MyProject.csproj" -c Release -o /app/publish /p:UseAppHost=false

FROM mcr.microsoft.com/dotnet/runtime:7.0 AS final
EXPOSE 5000
ENV ASPNETCORE_URLS=http://+:5000
WORKDIR /app
ENV PLAYWRIGHT_BROWSERS_PATH=/ms-playwright
COPY --from=publish /app/publish .
RUN mkdir /ms-playwright
COPY --from=base    /ms-playwright /ms-playwright 
ENTRYPOINT ["dotnet", "MyProject.dll"]
```
